src/Fixer/PhpUnit/PhpUnitTestCaseStaticMethodCallsFixer.php

```diff
@@ -0,0 +1,447 @@
+                ->setAllowedTypes(['string'])
+                ->setAllowedValues(array_keys(self::$allowedValues))
+                ->setDefault('static')
+                ->getOption(),
+            (new FixerOptionBuilder('methods', 'Dictionary of `method` => `call_type` values that differ from the default strategy.'))
+                ->setAllowedTypes(['array'])
+                ->setAllowedValues([static function ($option) {
+                    foreach ($option as $method => $value) {
+                        if (!isset(self::$staticMethods[$method])) {
+                            throw new InvalidOptionsException(
+                                sprintf(
+                                    'Unexpected "methods" key, expected any of "%s", got "%s".',
+                                    implode('", "', array_keys(self::$staticMethods)),
+                                    is_object($method) ? get_class($method) : gettype($method).'#'.$method
+                                )
+                            );
+                        }
+
+                        if (!isset(self::$allowedValues[$value])) {
+                            throw new InvalidOptionsException(
+                                sprintf(
+                                    'Unexpected value for method "%s", expected any of "%s", got "%s".',
+                                    $method,
+                                    implode('", "', array_keys(self::$allowedValues)),
+                                    is_object($value) ? get_class($value) : (null === $value ? 'null' : gettype($value).'#'.$value)
+                                )
+                            );
+                        }
+                    }
+
+                    return true;
+                }])
+                ->setDefault([])
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * @param Tokens $tokens
+     *
+     * @return int[][] array of [start, end] indexes from sooner to later classes
+     */
+    private function findPhpUnitClasses(Tokens $tokens)
+    {
+        $phpUnitTestCaseIndicator = new PhpUnitTestCaseIndicator();
+        $phpunitClasses = [];
+
+        for ($index = 0, $limit = $tokens->count() - 1; $index < $limit; ++$index) {
+            if ($tokens[$index]->isGivenKind(T_CLASS) && $phpUnitTestCaseIndicator->isPhpUnitClass($tokens, $index)) {
+                $index = $tokens->getNextTokenOfKind($index, ['{']);
+                $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $index);
+                $phpunitClasses[] = [$index, $endIndex];
+                $index = $endIndex;
+            }
+        }
+
+        return $phpunitClasses;
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $startIndex
+     * @param int    $endIndex
+     */
+    private function fixPhpUnitClass(Tokens $tokens, $startIndex, $endIndex)
+    {
+        static $conversionMap = [
```

Comment:
Please don't use a static variable.

Reply:
Moved to a class private variable.
Still static btw: class variables shouldn't have memory issues, have they?

Reply:
everything static is global and persistent, as PHP can't say would you use it laster or not.
non-static can be freed after instance is destroyed

Reply:
Static variables removed